### ISSUE TITLE
fix for generic methods that differ only in return type

### ISSUE THREAD

**Issue by andlang** [MakeAWishFoundation/SwiftyMocky#113]
I found a bug when generic methods differ only in a generic return type:

```swift
    func foo<T>(bar: String) -> Array<T>
    func foo<T>(bar: String) -> Set<T>
```
At least for that simple example I fixed it in #112. I’m not sure, if it still works, when the functions have several/other generic parameters.

**Comment by amichnia** [comment]
Merged PR, I will try to release fix in upcoming week.

**Comment by amichnia** [comment]
released in 2.0.0
